AlicerceTests/Extensions/OptionalTests.swift

```diff
@@ -0,0 +1,30 @@
+//
+//  OptionalTests.swift
+//  Alicerce
+//
+//  Created by Luís Afonso on 07/04/2017.
+//  Copyright © 2017 Mindera. All rights reserved.
+//
+
+import XCTest
+
+@testable import Alicerce
+
+final class OptionalTests: XCTestCase {
+
+    func testThen_UsingNonOptionalValue_ShouldUnwrapTheValue() {
+        let anOptionalString: String? = "😎"
+
+        anOptionalString.then { value in
+            XCTAssertTrue(anOptionalString! == value, "🚀 \(value) unwrapped ✅")
```

Comment:
Yeah, they can be a bit confusing at first, but then you get the hang of it 😉 

As for `XCTAssertTrue`, on these cases I simply use `XCTAssert` which is the same 😛